### DISCUSSION TITLE
How to use a Custom Queries ?

### DISCUSSION THREAD

**Discussion by Winniepoom** [Budibase/budibase#9880] [Help] [answered]
I made a custom queries look like this,
![q](https://user-images.githubusercontent.com/107398226/222893871-006a0d1a-a839-4232-9b5f-a29903bcaf05.png)

but I can't find where to run my custom queries. I look at data provider > data > and I can't find it in here any clue or where I miss it ?

**Comment by melohagan** [comment, marked as answer]
Hey @Winniepoom 

Only **Read** type queries can be used by a **Data provider**. (See the *Function* dropdown) under *Config*.

**Create/Update/Delete** would be used by the Execute Query action

**Reply by mike-hand** [reply]
Pages are not found.

**Reply by andz-bb** [reply]
@mike-hand sorry about that!

[Execute Query action](https://docs.budibase.com/docs/data-actions#execute-query)
[Custom queries](https://docs.budibase.com/docs/data-sources#custom-queries)

**Reply by mike-hand** [reply]
Thanks!
